typed/typed.go

```diff
@@ -24,16 +24,22 @@ import (
 	"sigs.k8s.io/structured-merge-diff/v4/value"
 )
 
+type ValidationOptions int
+
+const (
+	Loose ValidationOptions = iota
```

Comment:
Also I'm actually renaming this `AllowDuplicates` since .. that's a lot closer to what it actually is.